src/analyze/analyze-verify-util.c

```diff
@@ -72,6 +72,43 @@ int verify_prepare_filename(const char *filename, char **ret) {
         return 0;
 }
 
+static int find_unit(const char *p, char **ret) {
+        _cleanup_free_ char *a = NULL, *t = NULL;
+        int r;
+
+        assert(p);
+        assert(ret);
+
+        r = path_make_absolute_cwd(p, &a);
+        if (r < 0)
+                return r;
+
+        r = RET_NERRNO(access(a, F_OK));
+        if (r >= 0) {
+                *ret = TAKE_PTR(a);
+                return 0;
+        }
+
+        if (!unit_name_is_valid(p, UNIT_NAME_INSTANCE))
+                return r; /* return the original error code given by the above access(). */
```

Comment:
This passes my stripped-down test for `foo@bar.service`, but still causes a failure in Git's test suite. The difference there is that the command is actually `systemd-analyze verify systemd/user/git-maintenance@hourly.service`. And it looks like `unit_name_is_valid()` doesn't like the slashes in the filename.

Looking at `verify_prepare_filename()` above, you probably need to extract the basename, operate on that, then paste it back onto the directory. Something like peff/systemd@e883729e9fec15ed5c273b8700af475697810a0b, which seems to work for me.

(I wondered at first why we could not simply use `verify_prepare_filename()`, but I think it is going the _other_ way, filling in a template with a parameter).